### PR TITLE
The download is now interrupted when offline

### LINKED ISSUE #13250: Trying to save page offline always shows Downloading...
### Description
While testing #3784 found that when trying to save a page in offline mode, the downloader dosn't cancel or warn about being offline instead keeps showing `Downloading...` message

### Steps to Reproduce
  1. Clean install 0.21.11
  2. Load any site in a new tab and go offline
  3. Save page, doesn't cancel or error keeps showing downloading
  4. Go online, still doesn't change remains downloading.

**Actual result:**
saveoffline

**Expected result:**
Should error/cancel the download when offline

**Reproduces how often:**
100%

### Brave Version

**about:brave info:**

Brave | 0.21.11
-- | --
V8 | 6.4.388.41
rev | e146cd8
Muon | 5.0.5
OS Release | 10.0.16299
Update Channel | Beta
OS Architecture | x64
OS Platform | Microsoft Windows
Node.js | 7.9.0
Brave Sync | v1.4.2
libchromiumcontent | 64.0.3282.140

**Reproducible on current live release:**
Yes

### Additional Information
cc: @darkdh

### PR DESCRIPTION
**Pull request by aushakou**
Fix #13250

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

1. Load any website and go offline.
2. Save page

3. Before: Downloading process is running forever. No error is shown.

save-offline1-gif

3. After: Downloading process is interrupted

save-offline2-gif

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/filtering.js [hunk 1]
```diff
@@ -581,6 +581,7 @@ function updateDownloadState (win, downloadId, item, state) {
 
 function registerForDownloadListener (session) {
   var repaint = false
+  let networkConnected = false
```
Comment (NejcZdovc) [inline app/filtering.js]: I think this should be null, so that you will not hit `INTERRUPTED` by default

**Comment by aushakou** [comment]
@jasonrsadler @NejcZdovc 
Hello, I changed the initial value of `networkConnected` to be `true`, so it will not interrupt downloading when there is network connection. 
I tested it and it works for me.
Let me know what you think. 
Thanks

**Comment by jasonrsadler** [comment]
Code works great, @aushakou 👍 

@NejcZdovc would you confirm this when you can?

**Comment by bsclifton** [comment]
Thanks for the patch 😄 However, I'm going to close as the issue this fixes was closed

**Pull request closed without merging**